apps/prairielearn/src/lib/editors.ts

```diff
@@ -1116,6 +1116,40 @@ export class QuestionDeleteEditor extends Editor {
   }
 }
 
+export class QuestionBatchDeleteEditor extends Editor {
+  private questions: Question[];
+
+  constructor(params: BaseEditorOptions & { questions: Question[] }) {
+    super(params);
+
+    this.questions = params.questions;
+    this.description = `Delete questions ${this.questions.map((x) => x.qid).join(', ')}`;
+  }
+
+  async write() {
+    debug('QuestionBatchDeleteEditor: write()');
+
+    await this.questions.forEach(async (question) => {
+      if (question.qid) {
+        await fs.remove(path.join(this.course.path, 'questions', question.qid));
+        await this.removeEmptyPrecedingSubfolders(
+          path.join(this.course.path, 'questions'),
+          question.qid,
+        );
+      }
+    });
+
+    return {
+      pathsToAdd: this.questions
+        .map((question) =>
+          question.qid ? path.join(this.course.path, 'questions', question.qid) : undefined,
+        )
+        .filter((x) => x !== undefined), //ignore non-existent qids
```

Comment:
right now, we're not erroring on null qids, just ignoring. we could change that behavior.

Reply:
All non-deleted questions will have a non-null QID, so this should always be safe.

Reply:
Switched to assert not null structure in b6937b9 to maintain back-compatibility.